### PR TITLE
Akka.Cluster.Sharding: enable `prefer-oldest` by default on `Replicator`

### LINKED ISSUE #6973: Akka.Cluster.Sharding: duplicate shards / entities
**Version Information**
Version of Akka.NET? v1.5 - all versions including v1.5.13
Which Akka.NET Modules? Akka.Cluster.Sharding

**Describe the bug**

We have very rough and loose data on this right now, but it's been reported by multiple users including on https://github.com/petabridge/Akka.Persistence.Azure/issues/350 - looks like there could be something wrong with Akka.Cluster.Sharding in v1.5 that allows a `Shard` to be allocated more than once. This is our thread to investigate.

## If you have run into this issue, provide a full dump of your config here please

### PR DESCRIPTION
**Pull request by Aaronontheweb**
## Changes

Partially resolves #6973 - this change should have been made as part of https://github.com/akkadotnet/akka.net/pull/5146 originally

## Checklist

For significant changes, please ensure that the following have been completed (delete if not relevant):

* [x] This change follows the [Akka.NET API Compatibility Guidelines](https://getakka.net/community/contributing/api-changes-compatibility.html).
* [x] This change follows the [Akka.NET Wire Compatibility Guidelines](https://getakka.net/community/contributing/wire-compatibility.html).
* [x] I have [reviewed my own pull request](https://getakka.net/community/contributing/index.html#review-your-own-pull-requests).
* [x] Design discussion issue #6973, #5146

### CODE REVIEW

File: src/contrib/cluster/Akka.Cluster.Sharding/ClusterShardingGuardian.cs [hunk 1]
```diff
@@ -297,7 +297,7 @@ public sealed class StartProxy : INoSerializationVerificationNeeded
             });
         }
 
-        private ReplicatorSettings GetReplicatorSettings(ClusterShardingSettings shardingSettings)
+        internal static ReplicatorSettings GetReplicatorSettings(ClusterShardingSettings shardingSettings)
```
Comment (Aaronontheweb) [inline src/contrib/cluster/Akka.Cluster.Sharding/ClusterShardingGuardian.cs:300]: Made this `internal` in order to make it easy to test the _real_ `ReplicatorSettings` that we're passing into DData - straight from the source.

File: src/contrib/cluster/Akka.Cluster.Sharding.Tests/ClusterShardingConfigSpec.cs [hunk 2]
```diff
@@ -74,5 +74,21 @@ public void Should_cluster_sharding_settings_have_default_config()
...
+            var clusterShardingSettings = ClusterShardingSettings.Create(Sys);
+            var replicatorSettings = ClusterShardingGuardian.GetReplicatorSettings(clusterShardingSettings);
+            
+            replicatorSettings.Should().NotBeNull();
+            replicatorSettings.Role.Should().BeNullOrEmpty();
+            
+            // only populated when remember-entities is enabled
+            replicatorSettings.DurableKeys.Should().BeEmpty();
+            replicatorSettings.MaxDeltaElements.Should().Be(5);
+            replicatorSettings.PreferOldest.Should().BeTrue();
```
Comment (Aaronontheweb) [inline src/contrib/cluster/Akka.Cluster.Sharding.Tests/ClusterShardingConfigSpec.cs:91]: Detect whether or not `PreferOldest` is enabled.

File: src/contrib/cluster/Akka.Cluster.Sharding/reference.conf [hunk 3]
```diff
@@ -207,6 +207,9 @@ akka.cluster.sharding {
   distributed-data {
     # minCap parameter to MajorityWrite and MajorityRead consistency level.
     majority-min-cap = 5
+    
+    # ShardCoordinator is singleton running on oldest
+    prefer-oldest = on
```
Comment (Aaronontheweb) [inline src/contrib/cluster/Akka.Cluster.Sharding/reference.conf:212]: `prefer-oldest` should have been enabled for the `DDataShardCoordinator` when https://github.com/akkadotnet/akka.net/pull/5146 was originally introduced, but I overlooked it. This should play a key factor in https://github.com/akkadotnet/akka.net/issues/6973 - where `LWWRegister` data from the `DDataShardCoordinator` is going missing when the coordinator is pushing updates as it's exiting the cluster.

**Review by Aaronontheweb — COMMENTED** [review]
Detailed my changes

**Review by Arkatufus — APPROVED** [review]
LGTM

**Pull request merged**